File: SalesFlow.WebUI/src/features/notification/types/Notification.ts
import type { NotificationType } from "./NotificationType";

export interface Notification {
  id: number;

  title: string;

  message: string;

  type: NotificationType;

  isRead: boolean;

  entityName: string | null;

  entityId: number | null;

  createdDate: string;
}

File: SalesFlow.Entity/Entities/Notification.cs
using SalesFlow.Entity.Common;
using SalesFlow.Entity.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace SalesFlow.Entity.Entities
{
    public class Notification : BaseEntity
    {
        public string Title { get; set; } = null!;

        public string Message { get; set; } = null!;

        public NotificationType Type { get; set; }

        public bool IsRead { get; set; }

        public int UserId { get; set; }

        public AppUser User { get; set; } = null!;

        public string? EntityName { get; set; }

        public int? EntityId { get; set; }
    }
}
